### PR TITLE
[DEVOPS-1086]  fault injection:  log injection state changes

### PR DESCRIPTION
**Pull request by deepfire**
This adds logging for fault injection state transitions -- that is enablement / disablement of particular injections.

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: infra/src/Pos/Infra/InjectFail.hs [hunk 1]
```diff
@@ -71,14 +71,18 @@ data FInjects m     = FInjects
     , listFInjects   :: m [FInject]
     }
 
+logWarningLoud :: (CanLog m, HasLoggerName m) => String -> m ()
```
Comment (KtorZ) [inline infra/src/Pos/Infra/InjectFail.hs]: Waiting for the next patch `logWarningLouder` :trollface:

File: infra/src/Pos/Infra/InjectFail.hs [hunk 2]
```diff
@@ -71,14 +71,18 @@ data FInjects m     = FInjects
...
+logWarningLoud s = do
+  logWarning   "***"
+  logWarning $ "*** " <> T.pack s
+  logWarning   "***"
```
Comment (KtorZ) [inline infra/src/Pos/Infra/InjectFail.hs]: What about using newlines characters here? What if some other logs got interleaved in-between actually :thinking: ?

File: infra/src/Pos/Infra/InjectFail.hs (hunk 1 above)
Comment (deepfire) [reply infra/src/Pos/Infra/InjectFail.hs]: : -)

File: infra/src/Pos/Infra/InjectFail.hs (hunk 2 above)
Comment (deepfire) [reply infra/src/Pos/Infra/InjectFail.hs]: @KtorZ, done!

**Comment by deepfire** [comment]
@KtorZ, the CI failures look unrelated -- what do we do about it?

**Comment by KtorZ** [comment]
Known integration test failure being worked on:

```
  integration/TransactionSpecs.hs:102:22: 
  1) Transactions asset-locked wallets can receive funds and transactions are confirmed in index
       Falsifiable (after 1 test):
       
       not expected: 0
```

Not related to this PR. Merging acknowledging this.

**Pull request merged**